### DISCUSSION TITLE
MongoDB secure network access for Vercel dynamic IP addresses

### DISCUSSION THREAD

**Discussion by GKosheev** [vercel/vercel#8972] [Help] [answered]
Vercel uses dynamic IP addresses. To connect my MongoDB database to the project I have to turn on "Network Access" for all IPs. As it's bad for security, I wonder how exactly should I secure my DB.  Before writing this discussion, I read this article from Vercel: 
[how-to-allowlist-deployment-ip-address](https://vercel.com/guides/how-to-allowlist-deployment-ip-address)

I'm hosting my Next.js app on Vercel and I am not completely sure how this mentioned solution should work and be implemented:

> To properly secure your database, we recommend using a randomly generated password, stored as an [environment variable](https://vercel.com/docs/concepts/projects/environment-variables), at least 32 characters in length, and to rotate this password on a regular basis.

**Comment by swarnava** [comment, marked as answer]
Hi, Mongodb has a documentation to setup connection with Vercel: https://www.mongodb.com/docs/atlas/reference/partner-integrations/vercel/
